File: ParteC#/ProyectoProgramacion3/ProyectoProgramacion3Model/Model/reportes/ReporteReclamo.cs
using ProyectoProgramacion3Model.Model.reclamos;
namespace ProyectoProgramacion3Model.Model.reportes;

public class  ReporteReclamo : Reporte 
{
    public List<Reclamo> reclamos { get; set; } = new List<Reclamo>();
    public int cantidadReservas { get; set; }
    public int cantidadReclamos { get; set; }
    public double porcentajeIncidencias { get; set; }
    public int totalProcede { get; set; }
    public int totalNoProcede { get; set; }
    public int totalPendientes { get; set; }
    public ReporteReclamo(int idReporte, DateOnly fechaGeneracion, DateOnly fechaInicioFiltro, DateOnly fechaFinFiltro, List<Reclamo> reclamos,int totalProcede, int totalNoProcede, int totalPendientes) 
        : base(idReporte, fechaGeneracion, fechaInicioFiltro, fechaFinFiltro)
    {
        this.reclamos = reclamos;
        this.cantidadReservas = reclamos.Count; // Asumiendo que cada reclamo corresponde a una reserva
        this.cantidadReclamos = reclamos.Count;
        this.porcentajeIncidencias = (cantidadReclamos / (double)cantidadReservas) * 100;
        this.totalProcede = totalProcede;
        this.totalNoProcede = totalNoProcede;
        this.totalPendientes = totalPendientes;
    }
    public ReporteReclamo()
    {
    }
}

File: Backend/ParteJAVA/ProyectoPro3Modelo/src/main/java/pe/edu/pucp/proyectopro3/modelo/reportes/ReporteReclamo.java
package pe.edu.pucp.proyectopro3.modelo.reportes;

import pe.edu.pucp.proyectopro3.modelo.reclamos.Reclamo;

import java.util.Date;
import java.util.List;

public class ReporteReclamo extends Reporte{
    private List<Reclamo> detalleReclamos;
    private int cantidadReservas;
    private int cantidadReclamos;
    private double porcentajeIncidencias;
    private int totalProcede;
    private int totalNoProcede;
    private int totalPendientes;

    public ReporteReclamo() {

    }

    public List<Reclamo> getDetalleReclamos() {
        return detalleReclamos;
    }

    public void setDetalleReclamos(List<Reclamo> detalleReclamos) {
        this.detalleReclamos = detalleReclamos;
    }

    public int getCantidadReservas() {
        return cantidadReservas;
    }

    public void setCantidadReservas(int cantidadReservas) {
        this.cantidadReservas = cantidadReservas;
    }

    public int getCantidadReclamos() {
        return cantidadReclamos;
    }

    public void setCantidadReclamos(int cantidadReclamos) {
        this.cantidadReclamos = cantidadReclamos;
    }

    public double getPorcentajeIncidencias() {
        return porcentajeIncidencias;
    }

    public void setPorcentajeIncidencias(double porcentajeIncidencias) {
        this.porcentajeIncidencias = porcentajeIncidencias;
    }

    public int getTotalProcede() {
        return totalProcede;
    }

    public void setTotalProcede(int totalProcede) {
        this.totalProcede = totalProcede;
    }

    public int getTotalNoProcede() {
        return totalNoProcede;
    }

    public void setTotalNoProcede(int totalNoProcede) {
        this.totalNoProcede = totalNoProcede;
    }

    public int getTotalPendientes() {
        return totalPendientes;
    }

    public void setTotalPendientes(int totalPendientes) {
        this.totalPendientes = totalPendientes;
    }

    public ReporteReclamo(int idReporte, Date fechaGeneracion, Date fechaInicioFiltro, Date fechaFinFiltro, List<Reclamo> detalleReclamos, int cantidadReservas, int cantidadReclamos, double porcentajeIncidencias, int totalProcede, int totalNoProcede, int totalPendientes) {
        super(idReporte, fechaGeneracion, fechaInicioFiltro, fechaFinFiltro);
        this.detalleReclamos = detalleReclamos;
        this.cantidadReservas = cantidadReservas;
        this.cantidadReclamos = cantidadReclamos;
        this.porcentajeIncidencias = porcentajeIncidencias;
        this.totalProcede = totalProcede;
        this.totalNoProcede = totalNoProcede;
        this.totalPendientes = totalPendientes;



    }
}
